pkg/terra/contract_config_tracker.go

```diff
@@ -31,43 +31,51 @@ func NewContractTracker(spec OCR2Spec, jobID string, client *Client, lggr Logger
 	}
 
 	contract := Contract{
-		ChainID:     spec.ChainID,
-		JobID:       jobID,
-		ContractID:  addr,
-		terra:       client,
-		Transmitter: spec.TransmissionSigner,
-		data:        make(chan Events),
-		stop:        make(chan struct{}),
-		log:         lggr,
+		ChainID:         spec.ChainID,
+		JobID:           jobID,
+		ContractAddress: addr,
+		terra:           client,
+		Transmitter:     spec.TransmissionSigner,
+		stop:            make(chan struct{}),
+		log:             lggr,
 	}
 
-	// begin websocket subscription using terra.Subscribe
-	return &contract, contract.terra.Subscribe(context.TODO(), jobID, contract.ContractID, contract.data)
+	return &contract, nil
 }
 
 // Start creates a loop with handler for different event types
-func (ct *Contract) Start() {
-	for {
-		select {
-		case data := <-ct.data:
-			// process received data
-			for _, event := range data.Events {
-				if event == "wasm-set_config" {
-					ct.newConfig <- struct{}{}
+func (ct *Contract) Start() error {
+	// TODO:; timeout
+	txes, err := ct.terra.clientCtx.Client.Subscribe(context.TODO(), ct.JobID,
+		fmt.Sprintf("tm.event='Tx' AND execute_contract.contract_address='%s'", ct.ContractAddress))
```

Comment:
So it seems like the WS is used for the head tracker and catching new set config events...we could reduce the number of connections by removing it.
* libocr will poll for the configs anyway (so the WS isn't really necessary)
* we could also replace the head tracker with just reading from rpc or another endpoint

it would reduce the complexity of having to subscribe and unsubscribe

Reply:
ah yes that did cross my mind, but the ws support basically just came for free with the cosmos client so it was quicker to actually drop it in as is. Will take a look to see if I can swap it out now

Reply:
up to you! i thought it originally would be an additional endpoint, but it seems like we get subscriptions for free. i'm curious how they do it - polling? or maybe something else

...but i guess removing the subscribing parts would reduce the number of goroutines + sub/unsub pieces

Reply:
ah yep easy, done